src/browser.ts

```diff
@@ -171,81 +199,83 @@ export class GrpcClient {
-        cancelController = new AbortController();
-        cancelSignal = cancelController.signal;
-      }
-      const cancelHandler: CancelHandler = {
-        canceller: cancelController,
-        cancelRequested: false,
-      };
-      const headers = Object.assign({}, authHeader);
-      headers['Content-Type'] = 'application/x-protobuf';
-
-      const grpcFallbackProtocol = opts.protocol || 'https';
-      let servicePath = opts.servicePath;
-      if (!servicePath) {
-        if (service.options && service.options['(google.api.default_host)']) {
-          servicePath = service.options['(google.api.default_host)'];
-        } else {
-          callback(new Error('Service path is undefined'));
-          return;
-        }
-      }
-      let servicePort;
-      const match = servicePath.match(/^(.*):(\d+)$/);
-      if (match) {
-        servicePath = match[1];
-        servicePort = match[2];
-      }
-      if (opts.port) {
-        servicePort = opts.port;
-      } else if (!servicePort) {
-        servicePort = 443;
-      }
-      const protoNamespaces: string[] = [];
-      let currNamespace = method.parent;
-      while (currNamespace.name !== '') {
-        protoNamespaces.unshift(currNamespace.name);
-        currNamespace = currNamespace.parent;
-      }
-      const protoServiceName = protoNamespaces.join('.');
-      const rpcName = method.name;
-
-      const url = `${grpcFallbackProtocol}://${servicePath}:${servicePort}/$rpc/${protoServiceName}/${rpcName}`;
-
-      const fetch = isBrowser() ? window.fetch : nodeFetch;
-      fetch(url, {
-        headers,
-        method: 'post',
-        body: requestData,
-        signal: cancelSignal,
-      })
-        .then(response => {
-          return response.arrayBuffer();
-        })
-        .then(buffer => {
-          callback(null, new Uint8Array(buffer));
-        })
-        .catch(err => {
-          if (!cancelHandler.cancelRequested || err.name !== 'AbortError') {
-            callback(err);
-          }
-        });
-      return cancelHandler;
+      return [method, requestData, callback];
     }
-
     const serviceStub = service.create(serviceClientImpl, false, false);
     const methods = this.getServiceMethods(service);
 
     const newServiceStub = service.create(serviceClientImpl, false, false);
     for (const methodName of methods) {
       newServiceStub[methodName] = (req, options, metadata, callback) => {
-        const cancelHandler = serviceStub[methodName].apply(serviceStub, [
-          req,
-          callback,
-        ]) as CancelHandler;
+        const [method, requestData, serviceCallback] = serviceStub[
+          methodName
+        ].apply(serviceStub, [req, callback]);
+
+        let cancelController, cancelSignal;
+        if (typeof AbortController !== 'undefined') {
+          cancelController = new AbortController();
+          cancelSignal = cancelController.signal;
+        }
+        const cancelHandler: CancelHandler = {
```

Comment:
We don't need `CancelHandler` anymore since it all now happens in this function.